src/history/HistoryTests.cpp

```diff
@@ -662,8 +908,10 @@ TEST_CASE_METHOD(HistoryTests, "Full history catchup",
 
     std::vector<Application::pointer> apps;
 
-    std::vector<uint32_t> counts = {0, std::numeric_limits<uint32_t>::max(),
-                                    60};
+    std::vector<std::tuple<uint32_t, CatchupMetrics>> cfgs = {
+        {0, {1, 1, 64, 4, 4, 1, 0}},
```

Comment:
I think we should not hardcode numbers like this as it's impossible to know if they are correct or how to change them in the future.
Instead we should have them computed using formulas using expected behavior. For example: a bucketlist with 3 non-empty buckets should trigger 3 downloads

Reply:
Now that I think of it, if you were breaking down the "expected metrics" classes such that they map 1:1 to each *Work, then you should be able to construct each of those with high level parameters and the numbers broken up by Work item should be self-explanatory.

Reply:
I can create them form CatchupRange class - is that what you mean?

Reply:
I see that you did something closer to what I meant in `split CatchupMetrics into smaller structs` later, but it doesn't look like the new proposed split is enough to make those numbers less cryptic.

I think that the inputs specified at the test level should not be those numbers (in most cases), but instead be describing the high level steps that describes the scenario being tested.

Maybe a little more detail on what I am thinking can help.

Looking at `CatchupWork`, it spawns the following Work:
 * mGetHistoryArchiveStateWork;
 * mDownloadLedgersWork;
 * mVerifyLedgersWork;
 * mGetBucketsHistoryArchiveStateWork;
 * mDownloadBucketsWork;
 * mApplyBucketsWork;
 * mDownloadTransactionsWork;
 * mApplyTransactionsWork;

So what you need is a way to translate a given scenario into what it means for those spun up work items which then can translated into the metrics you have.

Example:
Scenario being tested - catchup complete to ledger #64 (a checkpoint), from a blank node - no failures.
we expect in this scenario to have the something like the following sequence:
 * mGetHistoryArchiveStateWork: archive queried once
 * mDownloadLedgersWork: should download the files to fetch ledger headers 3..64
 * mVerifyLedgersWork: should run once
 * mGetBucketsHistoryArchiveStateWork: should do nothing
 * mDownloadBucketsWork: should do nothing
 * mApplyBucketsWork: should do nothing
 * mDownloadTransactionsWork: should download files to cover ledger transitions 2->3, 3->4, …, 63->64
 * mApplyTransactionsWork: should close 62 ledgers
	
And from this, I think you should be able to compute the expected numbers.